projects/l2_pooling/noise_tolerance.py

```diff
@@ -0,0 +1,809 @@
+  plt.xticks([0.01 * n for n in xrange(0, 101, 10)])
+  plt.ylabel("Success rate")
+  plt.yticks([0.0, 0.2, 0.4, 0.6, 0.8, 1.0])
+  plt.title("Inference with normally distributed noise (stdev=0.1)")
+
+  plotPath = os.path.join("plots",
+                          "varyingNoise_successRate_varyProximalSampleSize_%s.pdf"
+                          % time.strftime("%Y%m%d-%H%M%S"))
+  plt.savefig(plotPath, bbox_extra_artists=(lgnd,), bbox_inches="tight")
+  print "Saved file %s" % plotPath
+
+
+def varyingNoise_varyProximalThreshold():
+  #
+  # Run the experiment
+  #
+  noiseLevels = [x * 0.01 for x in xrange(0, 101, 5)]
+  noiseSigma = 0.1
+  thresholds = [2, 8, 10, 15, 20]
+  numColumns = 3
+
+  results = defaultdict(list)
+
+  for trial in xrange(1):
+    print "trial", trial
+    objectDescriptions = createRandomObjectDescriptions(10, 10)
+
+    for minThresholdProximal in thresholds:
+      print "minThresholdProximal", minThresholdProximal
+      l2Overrides = {"minThresholdProximal": minThresholdProximal}
+      for noiseLevel in noiseLevels:
+        r = doVaryingNoiseExperiment(numColumns, l2Overrides,
+                                     objectDescriptions, noiseLevel, noiseSigma,
+                                     numInitialTraversals=6)
+        results[(minThresholdProximal, noiseLevel)].extend(r)
+
+  #
+  # Plot it
+  #
+  numCorrectActiveThreshold = 30
+  numIncorrectActiveThreshold = 10
+
+  plt.figure()
+  colorList = dict(zip(thresholds,
+                       ('r', 'k', 'g', 'b', 'y')))
+  markerList = dict(zip(thresholds,
+                        ('o', '*', 'D', 'x', 'o')))
+
+  for minThresholdProximal in thresholds:
+    y = []
+    for noiseLevel in noiseLevels:
+      trials = results[(minThresholdProximal, noiseLevel)]
+      numPassed = len([True for numCorrect, numIncorrect in trials
+                       if numCorrect >= numCorrectActiveThreshold
+                       and numIncorrect <= numIncorrectActiveThreshold])
+      y.append(numPassed / float(len(trials)))
+
+    plt.plot(noiseLevels, y,
+             color=colorList[minThresholdProximal],
+             marker=markerList[minThresholdProximal])
+
+  lgnd = plt.legend(["Proximal threshold %d" % minThresholdProximal
+                     for minThresholdProximal in thresholds],
+                    bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.0)
+  plt.xlabel("Mean feedforward noise level")
+  plt.xticks([0.01 * n for n in xrange(0, 101, 10)])
+  plt.ylabel("Success rate")
+  plt.yticks([0.0, 0.2, 0.4, 0.6, 0.8, 1.0])
+  plt.title("Inference with normally distributed noise (stdev=0.1)")
+
+  plotPath = os.path.join("plots",
+                          "varyingNoise_successRate_varyProximalThreshold_%s.pdf"
+                          % time.strftime("%Y%m%d-%H%M%S"))
+  plt.savefig(plotPath, bbox_extra_artists=(lgnd,), bbox_inches="tight")
+  print "Saved file %s" % plotPath
+
+
+if __name__ == "__main__":
+  constantNoise_varyColumns()
```

Comment:
Some other scripts demonstrate the same point (e.g., `constantNoise_varyProximalThreshold()` and `varyingNoise_varyProximalSampleSize()`), we can only need one of them here (also see my comments on adding a parameter of noiseSigma)